manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/ConnectorWorker.java

```diff
@@ -0,0 +1,206 @@
+        if (Objects.isNull(status)) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("Managed Connector status for '%s' [%s] is undetermined.",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+            }
+            return connectorEntity;
+        }
+        if (status.getState() == ConnectorState.READY) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("Managed Connector for '%s' [%s] is ready.",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+            }
+
+            // Connector is ready. We can proceed with the deployment of the Processor in the Shard
+            // The Processor will be provisioned by the Shard when it is in ACCEPTED state *and* Connectors are READY (or null).
+            return setReady(connectorEntity);
+        }
+
+        if (status.getState() == ConnectorState.FAILED) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("Creating Managed Connector for '%s' [%s] failed.",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+            }
+
+            // Deployment of the Connector has failed. Bubble FAILED state up to ProcessorWorker.
+            return setStatus(connectorEntity, ManagedResourceStatus.FAILED);
+        }
+
+        return connectorEntity;
+    }
+
+    private ConnectorEntity deployConnector(ConnectorEntity connectorEntity) {
+        // Creation is performed asynchronously. The returned Connector is a place-holder.
+        Connector connector = connectorsApi.createConnector(connectorEntity);
+        return setConnectorExternalId(connectorEntity, connector.getId());
+    }
+
+    @Transactional
+    protected ConnectorEntity setConnectorExternalId(ConnectorEntity connectorEntity, String connectorExternalId) {
+        ConnectorEntity resource = getDao().findById(connectorEntity.getId());
+        resource.setConnectorExternalId(connectorExternalId);
+        return resource;
+    }
+
+    @Override
+    protected ConnectorEntity runDeleteOfDependencies(Work work, ConnectorEntity connectorEntity) {
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(String.format("Destroying dependencies for '%s' [%s]",
+                    connectorEntity.getName(),
+                    connectorEntity.getId()));
+        }
+
+        // This is idempotent as it gets overridden later depending on actual state
+        connectorEntity = setStatus(connectorEntity, ManagedResourceStatus.DELETING);
+
+        Connector connector = connectorsApi.getConnector(connectorEntity);
+
+        // Step 1 - Clean up Kafka Topic if Connector cannot be found
+        if (Objects.isNull(connector)) {
+            return deleteTopic(connectorEntity);
+        }
+
+        // Step 1 - Clean up Kafka Topic if Connector is DELETED
```

Comment:
@manstis I'm wondering if on the delete, we delete the `Connector` before the topic. Basically we reverse the order of provisioning.

My concern is that if we delete the topic out from underneath the `Connector` it may enter the `FAILED` state and then we never try to delete it.